git/o-que-e-controle-de-versao.md

```diff
@@ -0,0 +1,16 @@
+# O que é controle de versão
+## Benefícios do controle de versão
+Sistemas de controle de versão são uma categoria de ferramentas de software que ajudam uma equipe de software a gerenciar mudanças ao longo do tempo.
+O software de controle de versão mantém todas as mudanças no código em um tipo especial de banco de dados. Se um erro for cometido, os desenvolvedores podem voltar no tempo e comparar versões anteriores do código para ajudar na correção minimizando a interrupção dos membros da equipe.
+
+Para quase todos os projetos de software, o código fonte é como as jóias da coroa - um recurso precioso cujo valor deve ser protegido. Para a maioria das equipes de software, o código fonte é um repositório de conhecimento e compreensão inestimáveis sobre o problema que os desenvolvedores coletaram e refinaram através de um esforço cuidadoso. Controle de versão protege o código fonte de catástrofe, erro humano e consequências não desejadas.
+
+Os desenvolvedores de software que trabalham em equipes estão continuamente escrevendo novos códigos e alterando os já existentes. O código para um projeto, aplicativo ou componente de software é normalmente organizado em uma estrutura de pastas ou "árvore de arquivos". Um desenvolvedor da equipe pode estar trabalhando em uma nova funcionalidade, enquanto outro corrige um bug que não tem relação com essa nova funcionalidade. Cada desenvolvedor pode fazer suas alterações em várias partes da árvore de arquivos.
+
+O controle de versão ajuda a equipe a resolver esses tipos de problemas, rastreando individualmente cada alteração feita por cada colaborador e ajudando a prevenir que o trabalho paralelo gere códigos conflitantes. As alterações feitas em uma parte do software podem ser incompatíveis com as feitas por outro desenvolvedor que trabalha ao mesmo tempo. Este problema deve ser descoberto e resolvido de forma ordenada sem gerar impedimentos para o trabalho do restante da equipe. Além disso, em desenvolvimento de software, qualquer alteração pode, por conta própria, introduzir novos bugs e o novo software não será confiável até que seja testado. Então o teste e o desenvolvimento caminham juntos até que uma nova versão esteja pronta.
+
+Um bom software de controle de versão suporta o workflow preferido do desenvolvedor sem impor uma forma específica de trabalhar. Idealmente, também funciona em qualquer plataforma, em vez de dizer qual sistema operacional ou ferramentas os desenvolvedores devem usar. Ótimos sistemas de controle de versão facilitam um fluxo de mudanças suave e contínuo no código, diferente do que acontece no frustrante e desajeitado mecanismo de bloqueio de arquivos, dando sinal verde a um colaborador, mas ao preço de bloquear o progresso dos outros.
```

Comment:
`Great version control systems facilitate a smooth and continuous flow` acredito que `smooth` nesse caso seja mais no sentido de "regular" do que "suave"